### PR TITLE
Locator strategy

### PR DESCRIPTION
**Pull request by AutomatedTester**
This pull request moves over all the remaining locator strategies to the current style of the specification.

### CODE REVIEW

File: webdriver-spec.html [hunk 1]
```diff
@@ -459,7 +460,15 @@
   <ul>
    <li><dfn data-lt="supported property index"><a href=http://heycam.github.io/webidl/#dfn-supported-property-indices>Supported property indices</a></dfn>
    <li><dfn><a href="https://heycam.github.io/webidl/#syntaxerror"><code>SyntaxError</code></a></dfn></li>
+   <!-- DOMException --><li><a href="https://heycam.github.io/webidl/#dfn-DOMException"><dfn><code>DOMException</code></dfn></a>
   </ul>
+
+  <dt>XPATH
+  <dd><p>The following terms are defined in the Document Object Model XPath standard [[!XPATH]]
```
Comment (andreastt) [inline webdriver-spec.html]: This whole section should go under the DOM section further up.

File: webdriver-spec.html [hunk 2]
```diff
@@ -3138,96 +3148,75 @@
...
-
   </ol>
-</section> <!-- /CSS Selector -->
 
-<p class=issue><strong>Warning:</strong>
- This rest of this section has not yet been redefined to match the <a>routing requests</a> model,
- and uses old concepts and definitions.
- Please do not rely on it yet.
+<h3>Link Text</h3>
+<p>To find a <a>WebElement</a> with the <dfn data-lt="link text selector">Link Text</dfn>
```
Comment (andreastt) [inline webdriver-spec.html]: WebElement → web element

File: webdriver-spec.html [hunk 3]
```diff
@@ -3138,96 +3148,75 @@
...
+   <a>strategy</a> the following steps need to be completed:
 
-<section>
-<h3>ECMAScript</h3>
+<ol>
+  <li><p>let <var>elements</var> be the result of calling <a>querySelectorAll</a>,
+    with argument <a>a elements</a>, with the <a>context object</a> equal to the
+    <var>start node</var>.
```
Comment (andreastt) [inline webdriver-spec.html]: let → Let, also applies in the CSS selector section above

File: webdriver-spec.html [hunk 4]
```diff
@@ -3138,96 +3148,75 @@
...
-            result = []
-            for element in elements:
-              text = element.text
-              if text == search_term:
-                result.append(element)
-          </pre>
-          <p>Where "search_term" is the link text being searched for, and "result" contains the list of elements to return.</p>
-    </div>
-</section> <!-- /Link Text -->
+<p>The <dfn data-lt="partial link text selector">Partial Link text</dfn> <a>strategy</a>
```
Comment (andreastt) [inline webdriver-spec.html]: Link → link

File: webdriver-spec.html [hunk 5]
```diff
@@ -3138,96 +3148,75 @@
...
+  is very similar to the <a>Link Text</a> <a>strategy</a>, but rather than matching
+  the entire string, only a subset needs to match. That is, this MUST return all
```
Comment (andreastt) [inline webdriver-spec.html]: subset → substring

File: webdriver-spec.html [hunk 6]
```diff
@@ -3138,96 +3148,75 @@
...
+  <a>a elements</a> with rendered text that partially or completely and case
```
Comment (andreastt) [inline webdriver-spec.html]: I think we should codify this to `<a><code>a</code> elements</a>` for clarity throughout.

File: webdriver-spec.html (hunk 5 above)
Comment (andreastt) [inline webdriver-spec.html]: No normative text in descriptions. This is what the algorithms are for.

File: webdriver-spec.html (hunk 6 above)
Comment (andreastt) [inline webdriver-spec.html]: “Partially or completely and case sensitively” is a bit much I think. It’s enough to say “… with rendered text that contains the selector expression”.

File: webdriver-spec.html [hunk 7]
```diff
@@ -3138,96 +3148,75 @@
...
-            result = []
-            for element in elements:
-              text = element.text
-              if text.search(search_term) != -1:
-                result.append(element)
-          </pre>
-          <p>Where "search_term" is the link text being searched for, and "result" contains the list of elements to return.</p>
-    </div>
-</section> <!-- /Partial Link Text -->
+        <li><p>If the result of <a>indexOf</a> on <var>rendered text</var> with argument
```
Comment (andreastt) [inline webdriver-spec.html]: I don’t think we should mandate the use of _indexOf_. Just say it’s a string that must contain something else. Implementations may not be in JS.

File: webdriver-spec.html (hunk 1 above)
Comment (AutomatedTester) [inline webdriver-spec.html]: These are not in the DOM spec, the DOM Spec doesnt deal with XPath as far as I can see.

**Comment by AutomatedTester** [comment]
fixed all comments and have landed.

**Pull request closed without merging**